src/lib/edf_schedule.c

```diff
@@ -442,6 +411,11 @@ static void schedule_edf(void)
 
 	/* no task to schedule */
 	spin_unlock_irq(&sch->lock, flags);
+
+	/* There is no prioritized task to handle at the moment.
+	 * Let's check if we have any idle task awaiting for execution.
+	 */
+	edf_schedule_idle();
```

Comment:
If prioritized task is detected then we switch over execution of idle tasks. This prioritized task may not be scheduled right away, but in e.g. some time, so we are gonna miss our idle window and not execute idle tasks now, but with the next interrupt. I think we should move call of `interrupt_set(PLATFORM_SCHEDULE_IRQ)` to above if and after that goto some label just before `edf_schedule_idle()`.

Reply:
> ...and not execute idle tasks now, but with the next interrupt
The idle task will be invoked *only* outside interrupt, on passive level so we won't wait until any interrupt to handle idle tasks. 

I am not sure if I understand your idea. You seems to be concerned about the the situation in which we detected prioritized task by `if (edf_task->state == SOF_TASK_STATE_QUEUED)` but this task deadline has not been reached yet so we skip its execution, therefore wasting time for idle tasks. I see it can happen but instead of setting IRQ which will trigger quite expensive context switch I would propose adding a check like: current = platform_timer_get(platform_timer);
		```
if (edf_task->state == SOF_TASK_STATE_QUEUED &&
		    edf_task->start <= current) {
goto schedule;
}
```
Would you agree?

Reply:
Yes, your idea is even better. Let's go with it.

Reply:
Good, updated.